rules/react.js

```diff
@@ -1,91 +1,146 @@
-		"react/jsx-closing-bracket-location": [2, "after-props"],
+		"react/jsx-closing-bracket-location": ["error", "after-props"],
 		// Don't allow spaces between curly braces in JSX attributes.
-		"react/jsx-curly-spacing": [2, "never"],
+		"react/jsx-curly-spacing": ["error", "never", {
+			"allowMultiline": true
+		}],
+		// Don't allow spaces before or after the equal sign for components.
+		"react/jsx-equals-spacing": ["error", "never"],
+		// Allow either .js or .jsx files for components.
+		"react/jsx-filename-extension": ["warn", {
+			"extensions": [".js", ".jsx"]
+		}],
+		// The first prop of a component should never be on a new line.
+		"react/jsx-first-prop-new-line": ["error", "never"],
 		// Ensure correct prefixing of event handlers in JSX.
-		"react/jsx-handler-names": [1, {
+		"react/jsx-handler-names": ["warn", {
 			"eventHandlerPrefix": "handle",
 			"eventHandlerPropPrefix": "on"
 		}],
 		// Require a tab indentation in props.
-		"react/jsx-indent-props": [2, "tab"],
+		"react/jsx-indent-props": ["error", "tab"],
 		// Require tab indentations in all JSX components
-		"react/jsx-indent": [2, "tab"],
+		"react/jsx-indent": ["error", "tab"],
 		// Require key props to be used.
-		"react/jsx-key": 2,
+		"react/jsx-key": "error",
 		// Limit number of props on a single line in JSX to 1
-		"react/jsx-max-props-per-line":[2, {
-			"maximum": 1
+		"react/jsx-max-props-per-line":["error", {
+			"maximum": 1,
+			"when": "always"
 		}],
-		// Allow arrow functions and binding because "premature optimization is the root of all evil"
-		// or at least because it messes with HMR. (h/t Donald Knuth via Gabe Isman)
-		"react/jsx-no-bind": 0,
+		// Require binding to happen in the constructor.
+		"react/jsx-no-bind": ["error", {
+			"allowArrowFunctions": true
+		}],
+		// Don't allow comments inside JSX statements.
+		"react/jsx-no-comment-textnodes": "error",
 		// Don't allow duplicate props in the same component
-		"react/jsx-no-duplicate-props": 2,
+		"react/jsx-no-duplicate-props": "error",
 		// Don't allow unwrapped strings
-		"react/jsx-no-literals": 2,
+		"react/jsx-no-literals": "error",
+		// Don't allow unsafe target=_blanks, must include rel='noreferrer noopener' too.
+		"react/jsx-no-target-blank": "error",
 		// Don't allow undeclared variables in JSX
-		"react/jsx-no-undef": 2,
+		"react/jsx-no-undef": "error",
 		// PascalCase for all components
-		"react/jsx-pascal-case": 2,
-		// Don't require prop types to be sorted in any particular order
-		"react/jsx-sort-prop-types": 0,
-		// Don't require props to be sorted in any particular order.
-		"react/jsx-sort-props": 0,
+		"react/jsx-pascal-case": "error",
+		// Don't require props to be sorted alphabetically, but put reserved props (key, ref) first,
+		// and callbacks last.
+		"react/jsx-sort-props": ["error", {
+			"noSortAlphabetically": true,
+			"reservedFirst": true,
+			"callbacksLast": true,
+		}],
```

Comment:
This is not necessary and provides no additional performance gains, but I believe it would be nice from a sanity perspective. Plus, keeping callbacks last is standard practice. More [here](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-sort-props.md).